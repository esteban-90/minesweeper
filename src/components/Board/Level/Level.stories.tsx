import type { Story, Meta } from '@storybook/react'
import { Level } from './Level'

export default {
  title: 'Components/Board/Level',
  component: Level,
} as Meta

const levels = ['beginner', 'intermediate', 'expert']

const Template: Story = (args) => <Level {...args}>{levels}</Level>

export const LevelExample = Template.bind({})
