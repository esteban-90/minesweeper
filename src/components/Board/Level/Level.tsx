import type { FC, PropsWithChildren } from 'react'
import styled from '@emotion/styled'

const Select = styled.select`
  margin: 0;
  border-radius: 0;
  border: 0.15vw solid;
  border-color: white #9e9e9e #9e9e9e white;
  background-color: #d1d1d1;
`

const Option = styled.option`
  font-weight: normal;
  diplay: block;
  white-space: nowrap;
  min-height: 1.2em;
  padding: 0 0.2vw 0.2vw;
`

export const Level: FC<PropsWithChildren> = ({ children }): JSX.Element => {
  return (
    <Select>
      {(children as string[]).map((item) => (
        <Option value={item} key={item}>
          {item}
        </Option>
      ))}
    </Select>
  )
}
