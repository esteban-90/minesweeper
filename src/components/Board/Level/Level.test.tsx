import { render } from '@testing-library/react'
import { Level } from './Level'

describe('Level test cases:', () => {
  it('should render correctly', () => {
    const levels = ['beginner', 'intermediate', 'expert']
    const { asFragment } = render(<Level>{levels}</Level>)

    expect(asFragment()).toMatchSnapshot()
  })
})
