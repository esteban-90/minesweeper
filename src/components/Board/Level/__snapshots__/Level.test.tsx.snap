// Vitest Snapshot v1

exports[`Level test cases: > should render correctly 1`] = `
<DocumentFragment>
  <select
    class="css-160ys9i"
  >
    <option
      class="css-1ycxzc0"
      value="beginner"
    >
      beginner
    </option>
    <option
      class="css-1ycxzc0"
      value="intermediate"
    >
      intermediate
    </option>
    <option
      class="css-1ycxzc0"
      value="expert"
    >
      expert
    </option>
  </select>
</DocumentFragment>
`;
